packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -967,6 +972,14 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             metadata,
         );
 
+        if (this.garbageCollector.shouldRunSweep) {
+            const expiryMs = this.runtimeOptions.gcOptions?.gcSessionExpiryTime ?? defaultContainerRuntimeExpiryMs;
```

Comment:
Similar comment: I think this value can't change for the lifetime of container instance. If that's true, then having runtimeOptions overwrite only works if it is remembered in a file when file is created and value from file is used in all future sessions. Otherwise, it has to be hard coded in package and never change.

I'm happy to be wrong here. But If I'm write, I'd prefer that flow of being able to change it over time as hard coding it forever seems like very limitting.

Reply:
This option is hidden and is added just for testing for now. We should not allow this to be changed.
In the future, we may want to change this value (say from 30 days to 45 days) but not via options but still hard coded in package. In that case, do we still want to store the value in the file so that existing sessions still use whatever value they value was when they were created? IMO, we may need to do this if we were to decrease the timeout in the future.

Reply:
Well, I think you imply that we will not change it (other than maybe initial increase). Are we sure? What if data suggests that there are really no sessions over 2 weeks, but users complain that files are large (and we know it's because of garbage)? Or vice versa, we learn that there are a ton of sessions over 30 days and people complain?

Actually, I think we will never be able to change it in either direction. We can increase it, but we will have to ensure that only newer clients (with longer duration) are summarizers, as if old client has lower timer to GC, but newer client lives longer, then we are screwed. Other way around I believe also gives us screwed state.

Reply:
I think a good first step is to add this timeout to the summary during container creation and then this value is not allowed to change throughout the lifetime of the document. If we realize that the timeout is less or more, we can at least change them for new files created after we release new bits.

For existing files, we will need to add some mechanism like code proposal or consensus like you suggested in the other comment.

Reply:
For existing files you can hard-code defaults and never change them. While it may not be ideal (and we get feedback that users do not like it), number of such files (%% wise) will decrease over time, so we might live with it (and punt on putting more expensive consensus thing).

I think it still leaves boolean flag in question. It has same problem - timers should be enabled for all clients in session or none. I.e., either we do sweep, and every single client knows about it, or we do not (across all clients).

We probably can add overwrite to kill whole feature (in case we learn things are not working as expected), but I'd expect this overwrite to force first client to overwrite file metadata, such that very quickly all clients are on the same page.